src/rules/no-duplicate-headings.js

```diff
@@ -29,46 +30,91 @@ export default {
 		messages: {
 			duplicateHeading: 'Duplicate heading "{{text}}" found.',
 		},
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					checkSiblingsOnly: {
+						type: "boolean",
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+
+		defaultOptions: [{ checkSiblingsOnly: false }],
 	},
 
 	create(context) {
-		const headings = new Set();
+		const [{ checkSiblingsOnly }] = context.options;
 		const { sourceCode } = context;
 
+		const headingsByLevel = checkSiblingsOnly
+			? new Array(7).fill(null).map(() => new Set())
+			: [null, new Set()];
```

Comment:
```suggestion
			? new Array(7).fill(null).map(() => new Set())
			: [null, new Set()];
```

Could you use a `Map` instead of an `Array`?

Currently, the leading `null` item in the array is unused and it causes some confusion.

Maybe we can use a `Map` like this:

```js
new Map([
  [1, []],
  [2, []],
  [3, []],
  [4, []],
  [5, []],
  [6, []]
])
```